packages/big-design/src/components/List/List.tsx

```diff
@@ -1,56 +1,25 @@
-import { Placement } from 'popper.js';
-import React, { memo } from 'react';
-import { Popper } from 'react-popper';
+import React, { useEffect, Ref } from 'react';
 
 import { StyledList } from './styled';
-import { ListPopperElement } from './ListPopperElement';
 
-interface Props {
-  handleListRef: React.RefObject<HTMLUListElement>;
+export interface ListProps extends React.HTMLAttributes<HTMLUListElement> {
   isOpen: boolean;
-  maxHeight?: number;
-  placement?: Placement;
-  positionFixed?: boolean;
+  maxHeight: number;
+  scheduleUpdate(): void;
 }
 
-type ListProps = Props & React.HTMLAttributes<HTMLUListElement>;
+interface PrivateProps {
+  forwardedRef: Ref<HTMLUListElement>;
+}
 
-export const List: React.FC<ListProps> = memo(
-  ({
-    children,
-    className,
-    handleListRef,
-    isOpen,
-    maxHeight = 250,
-    placement: selectedPlacement = 'bottom-start' as 'bottom-start',
-    positionFixed,
-    style,
-    ...rest
-  }) => (
-    <Popper
-      innerRef={handleListRef}
-      placement={selectedPlacement}
-      positionFixed={positionFixed}
-      modifiers={{ offset: { offset: '0, 10' } }}
-      eventsEnabled={isOpen}
-    >
-      {({ placement, ref, scheduleUpdate, style: popperStyle }) => (
-        <StyledList
-          data-placement={placement}
-          isOpen={isOpen}
-          maxHeight={maxHeight}
-          ref={ref}
-          style={popperStyle}
-          tabIndex={-1}
-          {...rest}
-        >
-          <ListPopperElement isOpen={isOpen} scheduleUpdate={scheduleUpdate}>
-            {children}
-          </ListPopperElement>
-        </StyledList>
-      )}
-    </Popper>
-  ),
-);
+const StyleableList: React.FC<ListProps & PrivateProps> = ({ forwardedRef, scheduleUpdate, ...props }) => {
+  useEffect(() => {
+    scheduleUpdate();
+  }, [React.Children.count(props.children), props.isOpen]);
```

Comment:
Wouldn't changing a `prop` on a `children` trigger the `useEffect`? I was thinking this would minimize the `useEffect` to only when the number of children changes, doesn't care about the props.

Reply:
There might be some instances where we do care about running the function when a child prop changes. It may be that the text of some item expands and drops onto a new line, then we would need to run the `scheduleUpdate` function.